File: ntier-template-web/src/app/auth/auth.models.ts
/**
 * Signed-in account profile returned by `/user/current`.
 */
export interface User
{
    id: number;
    email: string;
    firstName: string;
    lastName: string;
    displayName: string;
    roles: string[];
    initials: string;
    isAdmin: boolean;
}

/**
 * JWT pair returned from login and refresh.
 */
export interface TokenResponse
{
    accessToken: string;
    refreshToken: string;
    accessTokenExpiresAt: string;
}

/**
 * Token pair read from browser storage.
 */
export interface StoredTokens
{
    accessToken: string;
    refreshToken: string;
    accessTokenExpiresAt: string;
}

/**
 * Login request body for `POST /auth/login`.
 */
export interface LoginRequest
{
    email: string;
    password: string;
}

/**
 * Registration request body for `POST /auth/register`.
 */
export interface RegisterRequest
{
    email: string;
    password: string;
    displayName?: string;
    firstName?: string;
    lastName?: string;
}

/**
 * Email confirmation request body for `POST /auth/confirm-email`.
 */
export interface ConfirmEmailRequest
{
    userId: number;
    token: string;
}

/**
 * Resend confirmation request body for `POST /auth/resend-confirmation`.
 */
export interface ResendConfirmationRequest
{
    email: string;
}

/**
 * Forgot password request body for `POST /auth/forgot-password`.
 */
export interface ForgotPasswordRequest
{
    email: string;
}

/**
 * Reset password request body for `POST /auth/reset-password`.
 */
export interface ResetPasswordRequest
{
    email: string;
    token: string;
    newPassword: string;
}

/**
 * Refresh request body for `POST /auth/refresh`.
 */
export interface RefreshRequest
{
    refreshToken: string;
}

/**
 * Logout request body for `POST /auth/logout`.
 */
export interface LogoutRequest
{
    refreshToken?: string;
}

/**
 * Simple message response from anonymous auth endpoints.
 */
export interface MessageResponse
{
    message: string;
}

/**
 * Common API error payload shape.
 */
export interface ApiErrorBody
{
    message?: string;
    errors?: string[];
}

/**
 * Active authenticated session held in memory by {@link AuthService}.
 */
export interface AuthSession
{
    user: User;
    accessToken: string;
    refreshToken: string;
    accessTokenExpiresAt: string;
}

File: NTierTemplate.Api/Auth/AuthModels.cs
namespace NTierTemplate.Api.Auth;

/// <summary>
/// Login request payload.
/// </summary>
public class LoginRequest
{
    public required string Email { get; set; }

    public required string Password { get; set; }
}

/// <summary>
/// Registration request payload.
/// </summary>
public class RegisterRequest
{
    public required string Email { get; set; }

    public required string Password { get; set; }

    public string? DisplayName { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }
}

/// <summary>
/// Refresh request payload.
/// </summary>
public class RefreshRequest
{
    public required string RefreshToken { get; set; }
}

/// <summary>
/// Logout request payload.
/// </summary>
public class LogoutRequest
{
    public string? RefreshToken { get; set; }
}

/// <summary>
/// Email confirmation request payload.
/// </summary>
public class ConfirmEmailRequest
{
    public required int UserId { get; set; }

    public required string Token { get; set; }
}

/// <summary>
/// Resend email confirmation request payload.
/// </summary>
public class ResendConfirmationRequest
{
    public required string Email { get; set; }
}

/// <summary>
/// Forgot password request payload.
/// </summary>
public class ForgotPasswordRequest
{
    public required string Email { get; set; }
}

/// <summary>
/// Reset password request payload.
/// </summary>
public class ResetPasswordRequest
{
    public required string Email { get; set; }

    public required string Token { get; set; }

    public required string NewPassword { get; set; }
}

/// <summary>
/// Simple message response for anonymous auth actions.
/// </summary>
public class MessageResponse
{
    public required string Message { get; set; }
}

/// <summary>
/// Token response returned from login, register, and refresh.
/// </summary>
public class TokenResponse
{
    public required string AccessToken { get; set; }

    public required string RefreshToken { get; set; }

    public required DateTime AccessTokenExpiresAt { get; set; }
}
